moby/moby
# Unit tests for cli/command/image package

This PR adds unit tests for the cli/command/image package. Part of the work on #31217

Signed-off-by: Ignacio Capurro <icapurrofagian@gmail.com>

**- What I did**
Added unit tests for the following commands in package cli/command/image

- [x] history.go (100%)
- [x] inspect.go (100%)
- [x] load.go(100%)
- [x] pull.go (96.9%)
- [x] remove.go (100%)
- [x] tag.go (100%)
- [x] import.go (100%)
- [x] list.go (100%)
- [x] prune.go (100%)
- [x] push.go (95.2%)
- [x] save.go (100%)

To accomplish the 100% package coverage the following two commands need to be tested:
* build.go (issue link goes here)
* tag.go (issue link goes here)

I will the issues to work on them since this PR was getting really big and those two commands in particular will need some code refactor to accomplish a good coverage.

**- How I did it**
I followed @vdemeester previous test structure in other cli commands. Used gold files, builders created under `cli/internal/test/builders` and added new interfaces for some structs to be stubbed.

**- How to verify it**
Run the unit tests or check the Jenkins build. 

Before this PR
```
ok  	github.com/docker/docker/cli/command/image	0.086s	coverage: 0.0% of statements
```
After this PR
```
ok  	github.com/docker/docker/cli/command/image	0.346s	coverage: 45.4% of statements
```

**- Description for the changelog**

Added unit test coverage to some cli image commands.
